File: Practice/Language-Specific/Java/Stack-Java/Driver.java
/*
	Driver.java by Randyll Bearer 2018
	Practice for basic Java coding implementing a basic Stack data structure
	To be used in conjuction with Stack.java
*/


public class Driver {
	
	//Main driver method for the program, is in charge of instance'ing a Stack, and testing it's methods
	public static void main(String[] args){
		//Introductory Greeting
		System.out.println("Hello, now testing Stack.java...");
		System.out.println("\n\n\n");
		
		//Initialize a new Stack
		System.out.println("Initializing a new Stack...");
		Stack stack = new Stack();
		System.out.println("Stack Initialized");
		System.out.println("\n\n\n");
		
		//Test Pushing to Stack
		System.out.println("Pushing 1 to Stack...");
		stack.push(1);
		System.out.println("1 pushed to Stack");
		System.out.println("Pushing 2 to Stack...");
		stack.push(2);
		System.out.println("2 pushed to Stack");
		System.out.println("Pushing 3 to Stack...");
		stack.push(3);
		System.out.println("3 pushed to Stack");
		System.out.println("\n\n\n");
		
		//Retrieve data about the Stack
		System.out.println("Retrieving current capacity of Stack...");
		System.out.println("Current Stack Capacity = " + stack.getCapacity());
		System.out.println("Retrieving current numItems of Stack...");
		System.out.println("Current numItems in Stack = " + stack.getNumItems());
		System.out.println("\n\n\n");
		
		//Pop 2 items from Stack
		System.out.println("Popping item from Stack...");
		System.out.println(stack.pop() + " popped from Stack");
		System.out.println("Popping item from Stack...");
		System.out.println(stack.pop() + " popped from Stack");
		System.out.println("\n\n\n");
		
		//Retrieve data about the Stack
		System.out.println("Retrieving current capacity of Stack...");
		System.out.println("Current Stack Capacity = " + stack.getCapacity());
		System.out.println("Retrieving current numItems of Stack...");
		System.out.println("Current numItems in Stack = " + stack.getNumItems());
		System.out.println("\n\n\n");
		
	}
	
	
}
//End of File

File: Practice/Data Structures/Stack/Stack-PY/Driver.py
# Driver.py by Randyll Bearer 2018
# Practice for basic Python coding by testing a basic implementation of a Stack data structure
# To be used in conjuction with Stack.py

import Stack

class Driver:

	#Main driver method to test the Stack.py class/object
	def main():
	
		#Initializing the Stack
		print("Initializing the Stack...")
		stack = Stack.Stack()
		print("Stack Initialized.\n\n\n")
		
		#Adding contents to the Stack
		print("Adding contents to Stack...")
		print("Adding 1 to Stack...")
		stack.push(1)
		print("1 Added to Stack.")
		print("Adding 2 to Stack...")
		stack.push(2)
		print("2 Added to Stack.")
		print("Adding 3 to Stack...")
		stack.push(3)
		print("3 Added to Stack.\n\n\n")
		
		#Retrieve information from the Stack
		print("Getting numItems from Stack...")
		print(stack.getNumItems(), "items currently in Stack.\n\n\n")
		
		#Popping items from the Stack
		print("Popping items from Stack....")
		print("Popping top item from Stack...")
		print(stack.pop(), "popped from Stack.")
		print("Popping top item from Stack...")
		print(stack.pop(), "popped from Stack.")
		print("Popping top item from Stack...")
		print(stack.pop(), "popped from Stack.\n\n\n")
		
		#Retrieve information from the Stack
		print("Getting numItems from Stack...")
		print(stack.getNumItems(), "items currently in Stack.\n\n\n")
		
	if __name__ == '__main__':
		main()

#End of File